public/app/features/query/components/QueryGroup.tsx

```diff
@@ -87,6 +87,11 @@ export class QueryGroup extends PureComponent<Props, State> {
     });
 
     this.setNewQueriesAndDatasource(options);
+
+    // Clean up the first panel flag since the modal is now open
+    if (!!locationService.getSearchObject().firstPanel) {
+      locationService.partial({ firstPanel: null }, true);
+    }
```

Comment:
This already happens when adding the first panel into a dashboard, because you cannot edit a panel that doesn't exist. This URL is a hack to transport the state from the empty state component to another one (QueryGroup) to indicate you should open the advanced DS picker when mounted.

This is a temporal fix, the "onboarding" page with the modal open will be a different page to not open the Editor and not trigger unnecessary questions behind the modal. There is a task for this [#66946](https://github.com/grafana/grafana/issues/66946)